### PR TITLE
add truncate token filter docs #8450

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
add truncate token filter docs

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/8450

### Version
all

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _analyzers/token-filters/index.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ Normalization | `arabic_normalization`: [ArabicNormalizer](https://lucene.apache
 `synonym` | N/A | Supplies a synonym list for the analysis process. The synonym list is provided using a configuration file.
 `synonym_graph` | N/A | Supplies a synonym list, including multiword synonyms, for the analysis process.
 `trim` | [TrimFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/TrimFilter.html) | Trims leading and trailing white space from each token in a stream. 
-`truncate` | [TruncateTokenFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/TruncateTokenFilter.html) | Truncates tokens whose length exceeds the specified character limit. 
+[`truncate`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/truncate/) | [TruncateTokenFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/TruncateTokenFilter.html) | Truncates tokens whose length exceeds the specified character limit.
```
Comment (natebower) [inline _analyzers/token-filters/index.md:61]:
```suggestion
[`truncate`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/truncate/) | [TruncateTokenFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/TruncateTokenFilter.html) | Truncates tokens with lengths exceeding the specified character limit. 
```

File: _analyzers/token-filters/truncate.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
+---
+layout: default
+title: Truncate
+parent: Token filters
+nav_order: 440
+---
+
+# Truncate token filter
+
+The `truncate` token filter is used to shorten tokens that exceed a specified length. It trims tokens to a maximum number of characters, ensuring that tokens longer than this limit are truncated.
```
Comment (natebower) [inline _analyzers/token-filters/truncate.md:10]:
```suggestion
The `truncate` token filter is used to shorten tokens exceeding a specified length. It trims tokens to a maximum number of characters, ensuring that tokens exceeding this limit are truncated. 
```

File: _analyzers/token-filters/truncate.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+The `truncate` token filter is used to shorten tokens that exceed a specified length. It trims tokens to a maximum number of characters, ensuring that tokens longer than this limit are truncated. 
+
+## Parameters
+
+The `truncate` token filter can be configured with the following parameter.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`length` | Optional | Integer | Specifies the maximum length of the produced token. Default is `10`.
```
Comment (natebower) [inline _analyzers/token-filters/truncate.md:18]:
```suggestion
`length` | Optional | Integer | Specifies the maximum length of the generated token. Default is `10`.
```

File: _analyzers/token-filters/truncate.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+## Example
+
+The following example request creates a new index named `truncate_example` and configures an analyzer with `truncate` filter:
```
Comment (natebower) [inline _analyzers/token-filters/truncate.md:22]:
```suggestion
The following example request creates a new index named `truncate_example` and configures an analyzer with a `truncate` filter:
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Several changes. Thanks!

**Review by kolchfa-aws — APPROVED** [review]

**Pull request merged**